src/battle_util.c

```diff
@@ -11185,3 +11194,36 @@ bool32 IsGen6ExpShareEnabled(void)
     return FlagGet(I_EXP_SHARE_FLAG);
 #endif
 }
+
+bool8 CanTargetPartner(u8 battlerDef)
+{
+    return (gBattleTypeFlags & BATTLE_TYPE_DOUBLE)
+        && IsBattlerAlive(BATTLE_PARTNER(battlerDef))
+        && battlerDef != BATTLE_PARTNER(gBattlerAttacker);
+}
+
+bool8 TargetFullyImmuneToCurrMove(u8 battlerDef)
+{
+    u8 moveType = 0;
+    GET_MOVE_TYPE(gCurrentMove, moveType);
+
+    return (CalcTypeEffectivenessMultiplier(gCurrentMove, moveType, gBattlerAttacker, battlerDef, FALSE) & MOVE_RESULT_NO_EFFECT)
+         || IsBattlerProtected(battlerDef, gCurrentMove)
+         || IsSemiInvulnerable(battlerDef, gCurrentMove)
+         || DoesTargetHaveAbilityImmunity();
+}
+
+// For Dragon Darts
+static bool8 DoesTargetHaveAbilityImmunity(void)
+{
+    BattleScriptPushCursor(); // Backup the current script
+
+    if (AbilityBattleEffects(ABILITYEFFECT_MOVES_BLOCK, gBattlerTarget, 0, 0, 0)
+    || AbilityBattleEffects(ABILITYEFFECT_ABSORBING, gBattlerTarget, 0, 0, 0))
+    {
+        BattleScriptPop(); // Restore the original script
+        return TRUE;
+    }
+
+    return FALSE;
+}
```

Comment:
This terrifies me. Any mutation that occurs in the set up for a block/absorbing script will be applied during the accuracy check:
1. Straightforward mutation, e.g. [Soundproof](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/battle_util.c#L4972), [Flash Fire](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/battle_util.c#L5061), or [Earth Eater](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/battle_util.c#L5110).
2. Mutation that is expected to be undone by the script. I don't think there are any examples atm, but things like swapping attacker and target, with the intention that the script swaps them back.

I think in 1's case, a Fire-type Dragon Darts would silently boost a Flash Fire Pokémon's Fire-type moves and hit the non-Flash Fire Pokémon twice?

Reply:
Interestingly it hits the Flash Fire mon's ally once, which is even worse haha

Reply:
> I think in 1's case, a Fire-type Dragon Darts would silently boost a Flash Fire Pokémon's Fire-type moves and hit the non-Flash Fire Pokémon twice?

That's exactly what happens. Turned Dragon Darts into a Fire type move and it silently boosted the Flash Fire mon. 

I guess the best way to solve this would be to check all of the Immensities outside of `AbilityBattleEffects`?